src/bot/commands/command-play.ts

```diff
@@ -45,23 +25,47 @@ export class Play extends Command {
         guildId: String(voiceMember.guild.id),
       });
 
-      const player = this.getPlayer(voiceMember.channelId);
-      connection.subscribe(player);
+      const source = this.getSourceStream();
 
       const queue = this.getQueue();
-      queue.add(voiceMember.channelId, {
-        audioResource,
-        streamInfo,
-        userSearch: input,
+
+      const video = await source.getStreamFromUrl(input);
+
+      const searchResult =
+        video ??
+        ((await source.search(input, { limit: 1 })) as Array<StreamInfo>);
+
+      searchResult.forEach((streamInfo: StreamInfo) => {
+        queue.add(voiceMember.channelId, {
+          streamInfo: {
```

Comment:
you can pass through streamInfo directly if parameter name is the same
```suggestion
          queue.add (streamInfo, userSearch: input)
```

Reply:
resolved [82339b0](https://github.com/Marli-Marlete/marli-music/pull/21/commits/82339b069a67a8524fd04f2e85cc0c8f1f802384)